libsycl/include/sycl/detail/config.hpp

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the macros defining attributes for
+/// exported methods and defining API namespaces.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef __LIBSYCL_DETAIL_CONFIG_HPP
+#define __LIBSYCL_DETAIL_CONFIG_HPP
+
+#include <sycl/version.hpp>
+
+#define __LIBSYCL_BEGIN_UNVERSIONED_NAMESPACE namespace sycl {
+#define __LIBSYCL_END_UNVERSIONED_NAMESPACE }
+
+#define __LIBSYCL_BEGIN_VERSIONED_NAMESPACE                                    \
+  __LIBSYCL_BEGIN_UNVERSIONED_NAMESPACE inline namespace __LIBSYCL_ABI_NAMESPACE {
+#define __LIBSYCL_END_VERSIONED_NAMESPACE                                      \
+  }                                                                            \
+  __LIBSYCL_END_UNVERSIONED_NAMESPACE
+
+#ifndef __SYCL_DEVICE_ONLY__
+#ifndef __LIBSYCL_EXPORT
+#ifdef _WIN32
+
+#define __LIBSYCL_DLL_LOCAL
+
+#if __LIBSYCL_BUILD_SYCL_DLL
+#define __LIBSYCL_EXPORT __declspec(dllexport)
+#define __LIBSYCL_EXPORT_DEPRECATED(x) __declspec(dllexport, deprecated(x))
+#else
+#define __LIBSYCL_EXPORT __declspec(dllimport)
+#define __LIBSYCL_EXPORT_DEPRECATED(x) __declspec(dllimport, deprecated(x))
+#endif //__LIBSYCL_BUILD_SYCL_DLL
+
+#else // _WIN32
+
+#define __LIBSYCL_DLL_LOCAL __attribute__((visibility("hidden")))
```

Comment:
```suggestion
#define __LIBSYCL_DLL_LOCAL __attribute__((__visibility__("hidden")))
```

Is there a reason to use `__attribute__` instead of `[[]]`?